### PR TITLE
GROOVY-8778: Cast short-hand breaks for empty map

### PR DESCRIPTION
**Pull request by paulk-asert**

### CODE REVIEW

File: subprojects/parser-antlr4/src/main/java/org/apache/groovy/parser/antlr4/AstBuilder.java [hunk 1]
```diff
@@ -2337,14 +2337,11 @@ public Expression visitPathElement(PathElementContext ctx) {
...
-                MapEntryExpression mapEntryExpression = mapEntryExpressionList.get(0);
-
-                if (mapEntryExpression.getKeyExpression() instanceof SpreadMapExpression) {
-                    right = mapEntryExpression.getKeyExpression();
-                } else {
-                    right = mapEntryExpression;
-                }
+            if (mapEntryExpressionList.size() == 0) {
+                // expecting list of MapEntryExpressions later so use SpreadMap to smuggle empty MapExpression to later stages
+                right = new SpreadMapExpression(configureAST(new MapExpression(), ctx.namedPropertyArgs()));
```
Comment (danielsun1106) [inline subprojects/parser-antlr4/src/main/java/org/apache/groovy/parser/antlr4/AstBuilder.java]: It's better to `configureAST` the instance of `SpreadMapExpression` too, or its node position will be missing.

Comment (paulk-asert) [reply subprojects/parser-antlr4/src/main/java/org/apache/groovy/parser/antlr4/AstBuilder.java]: I had that originally but we throw that expression away in resolve visitor.

Comment (danielsun1106) [reply subprojects/parser-antlr4/src/main/java/org/apache/groovy/parser/antlr4/AstBuilder.java]: The implementation of parser should not decouple with back end(i.e. not rely on the implementation of back end). 
If the implementation of resolve visitor changes in the future, the expression is not thrown away, the node position is missing.

Comment (paulk-asert) [reply subprojects/parser-antlr4/src/main/java/org/apache/groovy/parser/antlr4/AstBuilder.java]: Yes, you are right - if we ever do support getAt(Map) more generally, we would need to process that differently, so best it is correct to start with.

**Pull request closed without merging**